cpp/src/spatial/lonlat_to_coordinate.cu

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <type_traits>
+#include <utility>
+#include <thrust/for_each.h>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cuspatial/error.hpp>
+
+namespace {
+
+using pair_of_columns = std::pair<std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>>;
+
+struct ll2coord_functor
+{
+    template <typename T, typename... Args>
+    std::enable_if_t<not std::is_floating_point<T>::value, pair_of_columns>
+    operator()(Args&&...)
+    {
+        CUSPATIAL_FAIL("Non-floating point operation is not supported");
+    }
+
+    template <typename T>
+    std::enable_if_t<std::is_floating_point<T>::value, pair_of_columns>
+    operator()(double origin_lon,
+               double origin_lat,
```

Comment:
Shouldn't origin_lon and origin_lat be type `T`, the same type as the column?

Also, from a performance standpoint, isn't there a substantial performance loss mixing 64 and 32 bit floats on the GPU?

Reply:
There's no loss from mixing. But 32-bit float operations are always faster than 64-bit. On some GeForce GPUs they are many times faster.